### PR TITLE
Updating dependencies in worker to support first rows from parquet

### PR DESCRIPTION
**Pull request by AndreaFrancis**
Preparation for https://github.com/huggingface/datasets-server/pull/988 implementation
As suggested in https://github.com/huggingface/datasets-server/pull/988#issuecomment-1487555308
I followed these steps:
- From main branch, ran make install
-  poetry remove apache-beam
- Updated the dependencies in pyproject.toml for datasets, hffs, pyarrow and numpy
- Ran manually poetry update for all the packages: datasets, hffs, pyarrow and numpy

### CODE REVIEW

**Comment by AndreaFrancis** [comment]
Weird but some tests for parquet-and-dataset-info job runner are failing, it looks like it is caused by pyarrow 11. I will investigate locally.

File: services/worker/tests/fixtures/hub.py [hunk 1]
```diff
@@ -375,8 +375,8 @@ def create_parquet_and_dataset_info_response(dataset: str, data_type: Literal["c
     }
 
 
-CSV_PARQUET_SIZE = 1_865
-AUDIO_PARQUET_SIZE = 1_383
+CSV_PARQUET_SIZE = 1_866
+AUDIO_PARQUET_SIZE = 1_384
```
Comment (AndreaFrancis) [inline services/worker/tests/fixtures/hub.py:379]: Increasing size to have test running successful. Looks like pyarrow 11 increased the size of computed parquet file.

**Review by severo — APPROVED** [review]
OK, thanks!

File: services/worker/tests/fixtures/hub.py [hunk 2]
```diff
@@ -328,7 +328,7 @@ def create_dataset_info_response_for_csv(dataset: str, config: str) -> Any:
     }
 
 
-def create_dataset_info_response_for_audio(dataset: str, config: str) -> Any:
+def create_dataset_info_response_for_audio() -> Any:
```
Comment (AndreaFrancis) [inline services/worker/tests/fixtures/hub.py:331]: Did this little refactor to make git actions alive again on this PR.

**Pull request merged**